packages/frontend/workspace/src/affine/index.ts

```diff
@@ -156,11 +209,9 @@ function setupAffineAwareness(
     );
   };
 
-  const awarenessUpdate = (changes: AwarenessChanges, origin: unknown) => {
-    if (origin === 'server') {
-      return;
-    }
-
+  // debounce awareness update to avoid too many updates
+  // BlockSuite current raise too many same awareness updates
+  const sync = debounce((changes: AwarenessChanges) => {
```

Comment:
Maybe we can exclude most of the broadcasts by caching the last sended awareness? In the old keck server, the broadcast number has been reduced by more than half just by comparing the last broadcast msg of the same user.

Reply:
edgeless mode will sync the cursor position with awareness, and debounce 500ms may make the remote cursor lag

Reply:
> Maybe we can exclude most of the broadcasts by caching the last sended awareness? In the old keck server, the broadcast number has been reduced by more than half just by comparing the last broadcast msg of the same user.

oh, that's cool. let's do it

Reply:
debounce with `500ms` may lower the cursor fps 👀